det3d/datasets/argoverse/argoverse_common.py

```diff
@@ -0,0 +1,330 @@
+    """Groups bounding boxes from five timestamps and performs the necessary SE3 transformation
+
+    Args:
+        box_dict: (dict) A dictionary with the timestamp as key and all bounding boxes in coordinate form as value
+        dataset: (str) A string pointing to the path of the Argoverse dataset
+        log_id: (str) The ID of the data collected
+        n_t: (int) Number of timestamps per group
+    Returns:
+        box_dict: (dict) A dictionary with the timestamp as key and five sets of bounding boxes from consecutive timestamps as value
+    """
+    curr_timestamps = []
+    for key in sorted(box_dict.keys()):
+        curr_timestamps.append(key)
+        if len(curr_timestamps) == n_t:
+            t0_to_map_SE3 = get_city_SE3_egovehicle_at_sensor_t(
+                int(curr_timestamps[0]), dataset, log_id
+            )
+            map_to_t0_SE3 = t0_to_map_SE3.inverse()
+            rotation = np.eye(3)
+            translation = np.array([XSize / 2, YSize / 2, 0])
+            t0_to_occ_SE3 = se3.SE3(rotation, translation)
+            new_labels = []
+            for ts in curr_timestamps:
+                labels_for_ts = box_dict[ts]
+                new_labels_for_ts = []
+                for label in labels_for_ts:
+                    new_label = Label(label.record, label.label_class)
+                    ego_to_map_SE3 = get_city_SE3_egovehicle_at_sensor_t(
+                        ts, dataset, log_id
+                    )
+                    new_label.boxes = ego_to_map_SE3.transform_point_cloud(label.boxes)
+                    new_label.boxes = map_to_t0_SE3.transform_point_cloud(
+                        new_label.boxes
+                    )
+                    new_label.boxes = t0_to_occ_SE3.transform_point_cloud(
+                        new_label.boxes
+                    )
+                    new_labels_for_ts.append(new_label)
+                new_labels.append(new_labels_for_ts)
+            box_dict[curr_timestamps[0]] = new_labels
+            curr_timestamps = curr_timestamps[1:]
+    for ts in curr_timestamps:
+        del box_dict[ts]
+    return box_dict
+
+
+def box_group_to_track_group(box_dict):
+    """Converts bounding box dictionary to list of track groups (boxes with the same track_id)
+    """
+    track_groups = {}
+    # k: timestamp; v: list of 5 lists, each being objects in that timestamp
+    for k, v in box_dict.items():
+        group_dict = {}
+        for idx, timestamp in enumerate(v):
+            for label in timestamp:
+                if label.record.track_id not in group_dict:
+                    group_dict[label.record.track_id] = {}
+                group_dict[label.record.track_id][idx] = label
+        group_list = [group_dict[k] for k in group_dict]
+        track_groups[k] = group_list
+    return track_groups
+
+def label_to_tensor(track_group_dict: Dict, n_t: int) -> Tuple[Dict, List]:
```

Comment:
can we state the types of the keys and values in the Dict, and the items in the List?